File: Apps/BankBusiness/backend/model/BusinessClient.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace backend.model;

[Table("business_client")]
public class BusinessClient
{

    [Key, Column(name: "id"), DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }
    
    [Column(name: "official_name")]
    public string OfficialName { get; set; }
    
    [Column(name: "brand")]
    public string Brand { get; set; }
    
    [Column(name: "email")]
    public string Email { get; set; }
    
    [Column(name: "password_hash")]
    public string PasswordHash { get; set; }

    public BusinessClient()
    {
    }

}

File: Apps/BankBusiness/backend/src/main/scala/model/BusinessClient.scala
package model

import io.getquill.{InsertMeta, SchemaMeta, UpdateMeta, insertMeta, schemaMeta, updateMeta}
import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder, JsonDecoder, JsonEncoder}
import zio.schema.{DeriveSchema, Schema}

case class BusinessClient(
                           id: Long,
                           officialName: String,
                           brand: String,
                           email: String,
                           passwordHash: String
                         ) {

}

inline given SchemaMeta[BusinessClient] = schemaMeta(
  "business_client",
  _.id -> "id",
  _.officialName -> "official_name",
  _.brand -> "brand",
  _.email -> "email",
  _.passwordHash -> "password_hash"
)

inline given InsertMeta[BusinessClient] = insertMeta(_.id)

inline given UpdateMeta[BusinessClient] = updateMeta(_.id)

implicit val businessClientSchema: Schema[BusinessClient] = DeriveSchema.gen[BusinessClient]

object BusinessClient {
  implicit val encoder: JsonEncoder[BusinessClient] = DeriveJsonEncoder.gen[BusinessClient]
  implicit val decoder: JsonDecoder[BusinessClient] = DeriveJsonDecoder.gen[BusinessClient]
}